poseidon2.nim

```diff
@@ -100,7 +100,20 @@ proc merkleRoot*(xs: openArray[F]) : F =
 
     return merkleRoot(ys)
 
-#-------------------------------------------------------------------------------
-
-
-
+proc unmarshal(
+        _: type F,
+        bytes: openArray[byte],
+        endian: static Endianness): seq[F] =
+  const chunkLen = 31
+  var elements: seq[F]
+  var i = 0
+  while i < bytes.len:
+    let chunk = bytes[i..<min(i + chunkLen, bytes.len)]
+    let bigint = B.unmarshal(chunk, endian)
```

Comment:
I'm not sure this `B.unmarshal` will do right thing with the last, broken chunk, in particular with the big-endian setting?

maybe a way to test this is to unmarshal both "0x123456" (3 bytes) and "0x00....00123456" (31 bytes) with big-endian setting?

Reply:
`B.unmarshal()` is from constantine, so it doesn't feel right to add tests for it here. I've made some tests in a branch on our fork of constantine, and they pass: https://github.com/codex-storage/constantine/pull/1.

Reply:
What I meant, is that I believe is that our unmarshal code behaves incorrectly, for the big-endian setting, when the input is not a multiple of `chunkLen`. And I proposed a way to sanity check this belief. But I don't have a working Nim+Constantine environment on my default machine (this will hopefully change at some point...) to actually do this.

Reply:
> I believe is that our unmarshal code behaves incorrectly,

I've added the sanity check that you proposed, and in the process moved the unmarshalling code into its own module, with its own tests. There did not appear to be anything wrong with the behaviour of the unmarshal code.

> I don't have a working Nim+Constantine environment

If you install Nim 1.6.16, then the only thing you need to do is run `nimble test` from the root of this repo; it will download constantine in the process.

Reply:
> If you install Nim 1.6.16, then the only thing you need to do is run nimble test 

Unfortunately constantine does not compile on my system, even with Nim 1.6.16 and doing the above.